scipy/signal/ltisys.py

```diff
@@ -1106,3 +1119,712 @@ def freqresp(system, w=None, n=10000):
+                _KNV0(B, ker_pole, transfer_matrix, i, poles)
+            else:
+                transfer_matrix_not_i_j = np.delete(transfer_matrix, (i, j),
+                                                    axis=1)
+                # after merge of gh-4249 great speed improvements could be
+                # achieved using QR updates instead of full QR in the line below
+                
+                #to debug with numpy qr uncomment the line below
+                #Q, _ = np.linalg.qr(transfer_matrix_not_i_j, mode="complete")
+                Q, _ = s_qr(transfer_matrix_not_i_j, mode="full")
+                
+                if np.isreal(poles[i]):
+                    if not np.isreal(poles[j]):
+                        msg = "mixing real and complex in YT_real" + str(poles)
+                        raise RuntimeError(msg)
+                    _YT_real(ker_pole, Q, transfer_matrix, i, j)
+                else:
+                    if np.isreal(poles[j]):
+                        msg = "mixing real and complex in YT_complex" + str(poles)
+                        raise RuntimeError(msg)
+                    _YT_complex(ker_pole, Q, transfer_matrix, i, j)
+
+        det_transfer_matrix = np.max((np.sqrt(np.spacing(1)),
+                                  np.abs(np.linalg.det(transfer_matrix))))
+        cur_rtol = np.abs(
+            (det_transfer_matrix -
+             det_transfer_matrixb) /
+            det_transfer_matrix)
+        if cur_rtol < rtol and det_transfer_matrix > np.sqrt(np.spacing(1)):
+            # Convergence test from YT page 21
+            stop = True
+        nb_try += 1
+    return stop, cur_rtol, nb_try
+
+
+def _KNV0_loop(ker_pole, transfer_matrix, poles, B, maxiter, rtol):
+    """
+    Loop over all poles one by one and apply KNV method 0 algorithm
+    """
+    # This method is useful only because we need to be able to call
+    # _KNV0 from YT without looping over all poles, otherwise it would
+    # have been fine to mix _KNV0_loop and _KNV0 in a single function
+    stop = False
+    nb_try = 0
+    while nb_try < maxiter and not stop:
+        det_transfer_matrixb = np.abs(np.linalg.det(transfer_matrix))
+        for j in range(B.shape[0]):
+            _KNV0(B, ker_pole, transfer_matrix, j, poles)
+
+        det_transfer_matrix = np.max((np.sqrt(np.spacing(1)),
+                                  np.abs(np.linalg.det(transfer_matrix))))
+        cur_rtol = np.abs((det_transfer_matrix - det_transfer_matrixb) /
+                       det_transfer_matrix)
+        if cur_rtol < rtol and det_transfer_matrix > np.sqrt(np.spacing(1)):
+            # Convergence test from YT page 21
+            stop = True
+
+        nb_try += 1
+    return stop, cur_rtol, nb_try
+
+
+def place_poles(A, B, poles, method="YT", rtol=1e-3, maxiter=30):
+    """
+    Compute K such as eigenvalues(A-dot(B, K))=poles.
+
+    K is the gain matrix such as the plant described by the linear system AX+BU
```

Comment:
Well I'm not a native english speakerbut I've seen that quite often, but what you suggest works too though. Just pick one and I'll change the docstring accordingly.